src/ctap/crypto_wrapper.rs

```diff
@@ -249,6 +260,10 @@ pub fn decrypt_credential_source(
     let sk_option = if credential_id.len() == LEGACY_CREDENTIAL_ID_SIZE {
         PrivateKey::new_ecdsa_from_bytes(&decrypted_id[..32])
     } else {
+        // Version number check.
+        if decrypted_id[4] == 0 {
```

Comment:
Shouldn't this be `!= 1`?

Reply:
I wasn't sure if having a best effort forward compatibility is desireable.

Reply:
For me, the fact that we go with version numbers, mean we don't expect any kind of polymorphism across versions, otherwise we would use CBOR maps. So I would say versions have to be check exactly.

Reply:
Changed. Also now that the version can be read before HMAC, this makes even more sense. Future versions might want stronger authenticity checks.
(Note that as an implementation detail, the HMAC is checked before the version, because all known versions need it. We are free to reorder in the future.)